### PR TITLE
Date range jsdocs updates

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-docs-update](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-docs-update)

File: src/app/public/modules/date-range-picker/types/date-range-relative-value.ts [hunk 1]
```diff
@@ -2,9 +2,11 @@ import {
   SkyDateRange
 } from './date-range';
 
-// Need to add the following to classes which contain static methods.
-// See: https://github.com/ng-packagr/ng-packagr/issues/641
-// @dynamic
+/** Need to add the following to classes which contain static methods.
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/date-range-picker/types/date-range-relative-value.ts:5]: Needs a newline after `/**`.

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/date-range-picker/types/date-range-relative-value.ts]: Done.

File: src/app/public/modules/date-range-picker/types/date-range-calculator.ts [hunk 2]
```diff
@@ -23,17 +23,33 @@ import {
...
   public readonly shortDescription: string;
+
+  /**
+   * Indicates the calculator type.
+   */
   public readonly type: SkyDateRangeCalculatorType;
 
   constructor(
+    /**
+     * Indicates the calculator id.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/date-range-picker/types/date-range-calculator.ts:39]:
```suggestion
     * Provides a calculator ID to specify calculator objects that represent date ranges.
```

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-docs-update](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-docs-update)

File: src/app/public/modules/date-range-picker/types/date-range-calculator.ts [hunk 3]
```diff
@@ -23,17 +23,33 @@ import {
...
 
 export class SkyDateRangeCalculator {
+
+  /**
+   * Indicates the text displayed within the calculator select menu.
+   */
   public readonly shortDescription: string;
+
+  /**
+   * Indicates the calculator type.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/date-range-picker/types/date-range-calculator.ts:33]:
```suggestion
   * Specifies the type of calculations available for the date range.
```
How do `calculatorId` and `type` work together? Looking at the lists for `SkyDateRangeCaluclatorId` and `SkyDateRangeCalculatorType`, it seems like users would use one or the other, but I'm not sure. Do we need to explain how these two properties work together?

File: src/app/public/modules/date-range-picker/types/date-range-calculator.ts [hunk 4]
```diff
@@ -23,17 +23,33 @@ import {
 } from './date-range';
 
 export class SkyDateRangeCalculator {
+
+  /**
+   * Indicates the text displayed within the calculator select menu.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/date-range-picker/types/date-range-calculator.ts:28]:
```suggestion
   * Specifies the text to display in the calculator select menu.
```
What is the calculator select menu? I get that this property is providing text to display somewhere, but I don't understand where it appears.

File: src/app/public/modules/date-range-picker/types/date-range-calculator.ts [hunk 5]
```diff
@@ -23,17 +23,33 @@ import {
...
     public readonly calculatorId: SkyDateRangeCalculatorId,
     private config: SkyDateRangeCalculatorConfig
   ) {
     this.type = config.type;
     this.shortDescription = config.shortDescription;
   }
 
+  /**
+   * Gets the current value of the calculator.
+   * @param startDateInput Indicates the start date value.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/date-range-picker/types/date-range-calculator.ts:50]:
```suggestion
   * @param startDateInput The start date.
```
Tweaking to follow the pattern of the other param descriptions on this page.

File: src/app/public/modules/date-range-picker/types/date-range-calculator.ts [hunk 6]
```diff
@@ -23,17 +23,33 @@ import {
...
+   * @param endDateInput Indicates the end date value.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/date-range-picker/types/date-range-calculator.ts:51]:
```suggestion
   * @param endDateInput The end date.
```

File: src/app/public/modules/date-range-picker/types/date-range-calculator.ts [hunk 7]
```diff
@@ -60,6 +76,10 @@ export class SkyDateRangeCalculator {
     };
   }
 
+  /**
+   * Performs synchronous validation against the control.
+   * @param value
```
Comment (blackbaud-johnly) [inline src/app/public/modules/date-range-picker/types/date-range-calculator.ts:81]: Do we need a description for this param?

File: src/app/public/modules/date-range-picker/types/date-range-calculator.ts [hunk 8]
```diff
@@ -23,17 +23,33 @@ import {
 } from './date-range';
 
 export class SkyDateRangeCalculator {
```
Comment (blackbaud-johnly) [inline src/app/public/modules/date-range-picker/types/date-range-calculator.ts:25]: Do we need a description to define `SkyDateRangeCalculator`?

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-docs-update](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-docs-update)

File: src/app/public/modules/date-range-picker/types/date-range-calculator.ts (hunk 4 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/date-range-picker/types/date-range-calculator.ts:28]: > What is the calculator select menu? I get that this property is providing text to display somewhere, but I don't understand where it appears.

Its the option/select menu that appears when consumers include the `sky-date-range-picker` component in their code.

File: src/app/public/modules/date-range-picker/types/date-range-calculator.ts (hunk 3 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/date-range-picker/types/date-range-calculator.ts:33]: `SkyDateRangeCalculatorType ` refers to the type of calculations that can be made. For example, a calculator with `calculatorId` of `After` uses a `SkyDateRangeCalculatorType.After`, while a `calculatorId` of `SpecificRange` uses `SkyDateRangeCalculatorType.Range`. I think the type is only important when you're creating your own custom calculator. It allows the consumer to define what type of calculation needs to be performed.

File: src/app/public/modules/date-range-picker/types/date-range-calculator.ts (hunk 7 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/date-range-picker/types/date-range-calculator.ts:81]: Oops. Nope. Class methods don't even show up in the docs anyways. Removed the param.

File: src/app/public/modules/date-range-picker/types/date-range-calculator.ts (hunk 8 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/date-range-picker/types/date-range-calculator.ts:25]: Not really sure what to say for this one. I was thinking it was kind of self-explanatory - it just represents the calculator. Any ideas?

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-docs-update](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-docs-update)

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Review by Blackbaud-DeniseCisneros — APPROVED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-docs-update](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-docs-update)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-docs-update](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-docs-update)

**Review by blackbaud-johnly — APPROVED** [review]

**Pull request merged**